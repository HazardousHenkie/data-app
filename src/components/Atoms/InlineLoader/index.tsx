import React from 'react'

import CircularProgress from '@material-ui/core/CircularProgress'
import LoaderWrapper from './styledComponents'

const InlineLoader: React.FC = () => {
    return (
        <LoaderWrapper>
            <CircularProgress color="inherit" />
        </LoaderWrapper>
    )
}

export default InlineLoader
