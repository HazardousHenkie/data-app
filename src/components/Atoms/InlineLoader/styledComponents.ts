import styled from 'styled-components'

export const LoaderWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`
