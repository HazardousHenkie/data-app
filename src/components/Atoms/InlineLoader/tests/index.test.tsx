import React from 'react'
import renderer from 'react-test-renderer'

import MockingAppComponent from 'utils/testing/MockingAppComponent'
import InlineLoader from '../index'

describe('<InlineLoader />', () => {
    it('should render like snapshot', () => {
        const component = renderer.create(
            <MockingAppComponent>
                <InlineLoader />
            </MockingAppComponent>
        )

        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
