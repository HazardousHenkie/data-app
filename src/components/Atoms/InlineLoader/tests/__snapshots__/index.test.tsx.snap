// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineLoader /> should render like snapshot 1`] = `
<div
  className="sc-AxjAm hsTYAM"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
